chain/client/cosmos.go

```diff
@@ -44,9 +50,32 @@ type CosmosClient interface {
 func NewCosmosClient(
 	ctx client.Context,
 	protoAddr string,
+	serverTlsCert string,
 	options ...cosmosClientOption,
 ) (CosmosClient, error) {
-	conn, err := grpc.Dial(protoAddr, grpc.WithInsecure(), grpc.WithContextDialer(dialerFunc))
+	var conn *grpc.ClientConn
+	var err error
+	if serverTlsCert == "" {
+		conn, err = grpc.Dial(protoAddr, grpc.WithInsecure(), grpc.WithContextDialer(dialerFunc))
+	} else {
+		rootCert, err := ioutil.ReadFile(serverTlsCert)
```

Comment:
The `NewCosmosClient` constructor shall be stateless. File checking and reading must be done in `main()` of the app. And the constructor must accept already initialised `RootCAs` as options.